app/models/case/sar/offender.rb

```diff
@@ -468,6 +468,10 @@ def prevent_number_change
     raise StandardError, "number is immutable" if current_state != "invalid_submission" && number_changed?
   end
 
+  def set_valid_case_number
```

Comment:
Created this new method rather than using the set_number private method. As it would need to moved to a public method and is used only for the case creation process. Thought it best to separate the two.